### ISSUE TITLE
Can't populate the dropdown field in Django autocomplete light

### ISSUE THREAD

**Issue by hosseinsa** [yourlabs/django-autocomplete-light#1180]
I am trying to create a dropdown of countires which can be searchable based on [this doc](https://django-autocomplete-light.readthedocs.io/en/master/tutorial.html), however almost disappointed to find any solution by googling.

I am using django-cities-light package and django-autocomplete-light, and all I need to do is creating a field of countries/cities and populating it in a form and making it searchable.

in setting.py and installed apps:

```
INSTALLED_APPS = [
    'dal',
    'dal_select2',
    'app.apps.MyownAppConfig',
    'users.apps.UsersConfig',
    'crispy_forms',
    'tinymce',
    'cities_light',
    'django.contrib.admin',
    'django.contrib.auth',
    ...
  ]
```

in models.py:

```
from tinymce.models import HTMLField
from cities_light.models import Country
class Post(models.Model):
     ...
     descript = HTMLField(blank=True, null=True)
     location = models.ForeignKey(Country, blank=True, null=True, on_delete=models.CASCADE)
     def __str__(self):
        return self.title


    def get_absolute_url(self):
        return reverse('post-detail', kwargs={'pk': self.pk})
```

in views.py:

```
class CountryAutocomplete(LoginRequiredMixin, autocomplete.Select2QuerySetView):
    def get_queryset(self):
        if not self.request.user.is_authenticated:
            return Country.objects.none()

        qs = Country.objects.all()

        if self.q:
            qs = qs.filter(name__icontains=self.q)

        return qs
```

In forms.py:

```
class postform(forms.ModelForm):

    class Meta:
        model = Post
        
        fields ='__all__'
        
        
        exclude = ['date_published']

        widgets = {
            'location': autocomplete.ModelSelect2(url='country-autocomplete' , attrs={'data-placeholder': 'select location...', 'data-minimum-input-length': 4})
        }

        search_fields = ['name']
```

and in postform.html:

```
{% extends "app/base.html" %}
{% load crispy_forms_tags %}
{% load static %}
{% block content %}
    <main>
     <form id="dropdownForm" method="POST" action="" enctype="multipart/form-data">
            <fieldset class="form-group">
                {% csrf_token %}
                <div class="content-section">

                    {{ form.descript }}
                    {{ form.location }}
                    <script type="text/javascript" src="http://dal- 
                        yourlabs.rhcloud.com/static/collected/admin/js/vendor/jquery/jquery.js"> 
                    </script>
                    
                    

                </div>
          </fieldset>
      </form>
   </main> 
{{ form.media }}
{% endblock content %}
```

and in url.py:

`urlpatterns = [path('country-autocomplete/', CountryAutocomplete.as_view(model=Country), name='country-autocomplete')]
`

But what I get is that location field is empty with -------- 

I also get this error in terminal:

> "GET /static/vendor/select2/dist/css/select2.css HTTP/1.1" 404 1832

which certainly mentioning such a file does not exits in subfolder of select2 in my venv!

**Comment by zak-ken** [comment]
I am getting a similar issue.

**Comment by frnmst** [comment]
I tried with the previous version (`3.5.0`) and it works. See #1172

**Comment by jpic** [comment]
Not really sure, can you try to make a patch ?

**Comment by yliharma** [comment]
Which version of Django are you using?

**Comment by jpic** [comment]
@yliharma I'm using the last one, but did you try with 3.7.0.dev0 ? It's the current master that I am currently releasing as 3.7.0, it might fix that too

**Comment by yliharma** [comment]
Yes I installed exactly that dev version, then some admin pages worked and other didn't, so I switched back to the 3.5.1 but also had to add to my Media classes the javascript files required by DAL in the correct order. 
It's basically a mess...the upgrade from version 2 to 3 of django autocomplete light had been a catastrophe but the upgrade from Django 1.11 to Django 3 is tricky as hell :(

**Comment by jpic** [comment]
Can you reproduce your issue in the test project of this repository ?

**Comment by jpic** [comment]
Also, try 3.8.0

Otherwise, use the alight branch of this repository, see if the autocomplete works on your page (it's not select2)

**Comment by jpic** [comment]
I confirm that the full test suite including all browser tests passes well, with 90% code coverage.

As such I would believe that the problem is a conflict with your setup.

Such conflicts are unavoidable without Shadow Root (like in React etc and now in every browser).

The way we currently do JS in Django is comparable to the way PHP used to expose all of its features from mysql to image generation in the same setup.

If you don't want conflicts, then you need to switch to [Django WebComponents](https://gitlab.com/yourlabs/djwc), and something like [Autocomplete-Light](https://github.com/yourlabs/autocomplete-light), but it's early because shadow dom has not been activated on that component:
     
- I didn't know all available features of CSS variables when I started the PoC, so I thought that this would stuck all my projects into the same design
- I kind of remember that Shadow Root prevented plain form POST from working,     
  but I'm not so sure about that and I can still try again ;)    

Basically the frontend situation is as we know it, but I believe the standards of WebComponents are a preventive solution even though I've been practicing for a couple of months or so.

**Comment by jpic** [comment]
@hosseinsa where did you get that from ?

```
                    <script type="text/javascript" src="http://dal- 
                        yourlabs.rhcloud.com/static/collected/admin/js/vendor/jquery/jquery.js"> 
                    </script>
```

That's really long gone (OpenShift 2)

**Comment by hosseinsa** [comment]
Django version which I am using is **3.0.7**

@jpic Thank you for your reply. I actually could solve the issue with doing a couple of things, like removing the `django-autocomplete-light` folder and again replace it with a new one. Then reordered .js files as below:

```
<script type="text/javascript" src="{% static 'admin/js/vendor/jquery/jquery.js' %}"></script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"</script>
<link href="{% static '/autocomplete_light/select2.css' %}" type="text/css" media="all" rel="stylesheet" />
<script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-beta.1/dist/js/select2.min.js"></script>
```

As I needed to move forward, I convinced myself that, this bug which was a common bug through the developers  occurs when there is a conflict between `.js` files.

**Comment by hosseinsa** [comment]
> @hosseinsa where did you get that from ?
> 
> ```
>                     <script type="text/javascript" src="http://dal- 
>                         yourlabs.rhcloud.com/static/collected/admin/js/vendor/jquery/jquery.js"> 
>                     </script>
> ```
> 
> That's really long gone (OpenShift 2)

I really don't remember where, but most probably copied from the `stackoverflow`

**Comment by jpic** [comment]
@hosseinsa defnitely remove this line, can you try again with DAL 3.8.1 ? It should work

**Comment by jpic** [comment]
@hosseinsa because that could be the cause of loading several instances of jquery.

It's a problem that's gone in 3.8.1 anyway.

If you select boxes still aren't populated: can you confirm the select box works when you remove the autocomplete widget ?

In the generated HTML, what's the select field and children code like ?

**Comment by hosseinsa** [comment]
@jpic thanks for explanation. I updated to 3.8.1 and commented autocomplete widget so I see that the select box works.
This is the part of html code, for one of forms which I used:

<div class="form-group mb-5 mb-md-0">
     <select name="city" data-placeholder="<span class=&quot;fe fe-map-pin&quot;></span> Search by location..." data
                  html="" style="border-right:none;border-left:none;border-top:none;" required="" 
                  id="id_city" 
                 data-autocomplete-light-language="en" data-autocomplete-light-url="/app/ajax/location-autocomplete" data- 
                autocomplete-light-function="select2">
                <option value="" selected="">---------</option>

      </select>
</div>

**Comment by jpic** [comment]
:tophat:
